packages/core/handle/handle.js

```diff
@@ -676,10 +676,10 @@ const call = function (method) {
  * ```
  *
  * @method   adaptEvent
- * @param    {EventAdapter}  adapter  Function to adapt the event payload
- * @param    {Function}  handler  Handler to call with the new event payload
+ * @param    {EventAdapter}     adapter  Function to adapt the event payload
+ * @param    {HandlerFunction}  handler  Handler to call with the handler function
  *
- * @returns  {HandlerFunction}    New event payload
+ * @returns  {HandlerFunction}           Returns value of the handler function
```

Comment:
Maybe the description should be closer to the description of the return from `oneOf`.  Maybe something like:
Returns a function that accepts an event (to be modified) and props and returns the return value of `handler`.
Or
Returns an [event handler]{@link core/handle.HandlerFunction} (suitable for passing to `handle`) that returns the result of `handler`
Something like that?